### PR TITLE
refactor(vue-button): use composition api + re-design

### PR DESCRIPTION
**Pull request by devCrossNet**
related to #487

### What is accomplished by your PR?
This PR updates:
- the design system typo and colors
- uses composition API for the button component
- re-designs the button component according to the new designs

Bildschirmfoto 2020-01-06 um 00 48 22

### Is there something controversial in your PR?
no

### Link to the Issue
#487 

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR
- [ ] Change documentation for the code introduced by your PR
- [x] Add Story / Design System Page for a new component introduced by your PR

### CODE REVIEW

File: src/app/shared/designSystem/variables/_color-palette.scss [hunk 1]
```diff
@@ -1,64 +1,46 @@
...
-  'green-20': #c6f7c6,
-  'green-30': #90e0a4,
-  'green-40': #40bf77,
-  'green-50': #0d9660,
-  'green-60': #096e4c,
-  'green-70': #045a3f,
-  'green-80': #004934,
-  'green-90': #003c2b,
-  'green-100': #003023,
+  'coral-30': #eab3c2,
```
Comment (devCrossNet) [inline src/app/shared/designSystem/variables/_color-palette.scss]: @einfalles this is the new naming for the color palette, please review 😊

File: src/app/shared/designSystem/variables/_typo.scss [hunk 2]
```diff
@@ -11,6 +11,22 @@ $font-size-h4: 2rem;
 $font-size-h5: 1.6rem;
 $font-size-h6: 1.4rem;
 
+$font-size-xxl: 4.8rem;
```
Comment (devCrossNet) [inline src/app/shared/designSystem/variables/_typo.scss]: @einfalles let's talk about the naming convention for sizes here

**Comment by devCrossNet** [comment]
@einfalles let me know if you want to check it locally and you're not sure how to set it up

**Comment by arjamizo** [comment]
I dislike that refactor and re-design takes place in single PR, I think this has to change

I would like to see full refactor to composition API, ideally with a fallback in a separate file to an old-fashion vue architecture 

and after that I think there should be any redesigns

**Comment by devCrossNet** [comment]
@arjamizo can you elaborate a little bit? Are you concerned about migration to the new version?

maybe, I explain why I chose to do it that way. In version 4 we want to roll out the new design-system + vue.js 3 and the benefits of code sharing + new testing best practices by using the testing-library.

So, I have to refactor all components 3 times if I would split that up. Since I don't have enough time I go through all components once and do the 3 tasks together in order to save time.

I can change this if there is a good reason. Otherwise, I would favor time over documentation.

**Pull request merged**